react-accordion-component/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+
+type Items = {
+  id: number;
+  title: string;
+  content: string;
+};
+
+export function Accordion({ arr }: { arr: Items[] }) {
+  const [id, setId] = useState(0);
+
+  const item = arr.map((i) => (
+    <RenderTopic item={i} key={i.id} id={id === i.id} setId={setId} />
+  ));
+  return item;
+}
+
+type List = {
+  item: Items;
+  id: boolean;
+  setId: (num: number) => void;
+};
+
+function RenderTopic({ item, id, setId }: List) {
+  return (
+    <>
+      <button
+        onClick={() => {
+          if (id === true) {
```

Comment:
I would move this logic up to the parent. Don't put so much logic in your subcomponents. Keep it up in the parent.